lib/paper_trail.rb

```diff
@@ -47,32 +48,46 @@ def enabled?
       !!PaperTrail.config.enabled
     end
 
-    # Sets whether PaperTrail is enabled or disabled for the current request.
+    # Returns the PaperTrail::Request object for setting keys for a single request
     # @api public
-    def enabled_for_controller=(value)
-      paper_trail_store[:request_enabled_for_controller] = value
-    end
-
-    # Returns `true` if PaperTrail is enabled for the request, `false` otherwise.
-    #
-    # See `PaperTrail::Rails::Controller#paper_trail_enabled_for_controller`.
-    # @api public
-    def enabled_for_controller?
-      !!paper_trail_store[:request_enabled_for_controller]
+    def request(options = nil)
+      if options
+        old_request = @request
+        @request ||= PaperTrail::Request.new
+        yield
+        @request = old_request
+      else
+        @request ||= PaperTrail::Request.new
+      end
     end
 
-    # Sets whether PaperTrail is enabled or disabled for this model in the
-    # current request.
-    # @api public
-    def enabled_for_model(model, value)
-      paper_trail_store[:"enabled_for_#{model}"] = value
+    # All methods that cause changes to the keys that affect the current
+    # request are now moved into PaperTrail::Request
+    REQUEST_STORE_DEPRECATED_METHODS = %i[
+      enabled_for_controller=
+      enabled_for_controller?
+      enabled_for_model
+      enabled_for_model?
+      whodunnit=
+      whodunnit
+      controller_info=
+      controller_info
+      transaction_id
+      transaction_id=
+    ].freeze
+
+    def method_missing(method, *args, &block)
+      if REQUEST_STORE_DEPRECATED_METHODS.include?(method)
+        msg = format("Use paper_trail.request.%s instead of paper_trail.%s", method, method)
+        ::ActiveSupport::Deprecation.warn(msg, caller)
```

Comment:
I know I said " .. the old methods could eventually be deprecated" but let's not do it yet.

Reply:
:+1: